site/gatsby-site/src/components/forms/IncidentForm.js

```diff
@@ -0,0 +1,535 @@
+  text: Yup.string()
+    .min(80, '*Text must have at least 80 characters')
+    .max(50000, "*Text can't be longer than 50000 characters")
+    .required('*Text is required'),
+  incident_date: Yup.string()
+    .matches(dateRegExp, '*Date is not valid, must be `YYYY-MM-DD`')
+    .required('*Incident date required'),
+  date_published: Yup.string()
+    .matches(dateRegExp, '*Date is not valid, must be `YYYY-MM-DD`')
+    .required('*Date published is required'),
+  date_downloaded: Yup.string()
+    .matches(dateRegExp, '*Date is not valid, must be `YYYY-MM-DD`')
+    .required('*Date downloaded required'),
+  url: Yup.string()
+    .url('*Must enter URL in http://www.example.com format')
+    .required('*URL required'),
+  image_url: Yup.string().matches(
+    /((https?):\/\/)(\S)*$/,
+    '*Must enter URL in http://www.example.com/images/preview.png format'
+  ),
+  incident_id: Yup.number().integer('*Must be an incident number or empty'),
+});
+
+const defaultValue = {
+  title: '',
+  authors: '',
+  submitters: 'Anonymous',
+  incident_date: '',
+  date_published: '',
+  date_downloaded: '',
+  url: '',
+  image_url: '',
+  incident_id: '',
+  text: '',
+};
+
+const IncidentForm = ({ incident }) => {
+  const { loading, user } = useUser();
+
+  const [csvData, setCsvData] = useState([{ data: incident || defaultValue }]);
+
+  const [csvIndex, setCsvIndex] = useState(0);
+
+  const [relatedIncidents, setRelatedIncidents] = useState([]);
+
+  const isAdmin = !loading && user.type === 'token';
+
+  const handleOnDrop = (data) => {
+    console.log('---------------------------');
+    console.log(data);
+    setCsvData(data);
+    console.log('---------------------------');
+  };
+
+  const handleOnError = (err, file, inputElem, reason) => {
+    console.log(err, file, inputElem, reason);
+  };
+
+  const updateRelated = (incident_id) => {
+    const parsed = parseInt(incident_id);
+
+    console.log(incident_id);
+    if (isNaN(parsed)) {
+      setRelatedIncidents([]);
+      return;
+    }
+
+    runQuery(
+      { incident_id: parsed },
+      setRelatedIncidents,
+      config['realm']['production_db']['db_service'],
+      config['realm']['production_db']['db_name'],
+      config['realm']['production_db']['db_collection']
+    );
+  };
+
+  const previousRecord = (setValues) => {
+    const newCSVIndex = Math.max(0, csvIndex - 1);
+
+    setValues(csvData[newCSVIndex]['data']);
+    setCsvIndex(newCSVIndex);
+  };
+
+  const nextRecord = (setValues) => {
+    const newCSVIndex = Math.min(csvData.length - 1, csvIndex + 1);
+
+    setValues(csvData[newCSVIndex]['data']);
+    setCsvIndex(newCSVIndex);
+  };
+
+  const onSubmit = async (values, { setSubmitting, resetForm }) => {
+    // When button submits form and form is in the process of submitting, submit button is disabled
+    setSubmitting(true);
+
+    if (incident) {
+      // Do update
+      // await user.functions.updateReportInReview(values)
```

Comment:
@smcgregor , here we update the incident, can you please prepare the updateFunction so I can make a call here?